designs/air-gapped.md

```diff
@@ -0,0 +1,97 @@
+# Air Gapped user experience consolidation
+
+## Introduction
+
+**Problem:** we offer a set features that allow users to run EKS-A cluster in disconnected environments, but they lack cohesion.
+
+Overtime, we have developed a particular set of features and tools that, even though they were aimed to solve independent problems, they all converge to offer to support for running EKS-A in environments without Internet connection.
+As a result of their independent development, their interfaces and behaviors are not in sync, which makes the user experience not ideal.
+
+This document aims to solve that, learning from the experience gained building such tools and offering a more cohesive and simple to use solution, without drastically changing the backbone of those features and how they interact with each other.
+
+### Tenets
+
+* ****Simple:**** minimal interface, minimum number of manual steps.
+
+### Goals and Objectives
+
+As a EKS-A cluster administrator I want to:
+
+* Manage clusters in an environment without Internet connection.
+* Populate my environment with the necessary dependencies without having to worry about what those dependencies are and with minimal effort.
+* Interact with EKS-A clusters in air-gapped environments in the same way I interact with any other EKS-A cluster.
+
+### Statement of Scope
+
+**In scope**
+* Download dependencies and package them.
+* Populate disconnected environments with dependencies without Internet connection.
+
+**Not in scope**
+* Setup additional infrastructure to support disconnected environments (eg. private registries).
+
+**Future scope**
+* Package dependencies selectively.
+* Populate disconnected environments with dependencies with Internet connection.
+ 
+## Current state
+In order to support disconnected environments, users currently have to:
+1. Provision an OCI registry.
+1. Run `eksctl anywhere download artifacts` to download manifests to disk.
```

Comment:
Late to the party here, does artifacts include OVAs?

Reply:
The current implementation of `download commands` doesn't download them, it only downloads yaml manifests

The new command I'm proposing, `export artifacts`, doesn't either, at least in this first proposal.

But it could definitely be expanded to include them. If we think we are gonna keep adding more types of artifacts, I was thinking that the tarball could contain a manifest with all the "exported" artifacts.

We would just need to define what's the behavior for `export artifacts`: do we upload them to a vcenter? or just untar them to disk?. But I would leave that for a different proposal when the times comes.

Reply:
Are you suggesting that download/export is the difference between manifests/tarball? That sounds a bit confusing to me too, unless we are planning to deprecate?

Reply:
Yeah, I'm proposing to deprecate the old ones